### ISSUE TITLE
Can't catch IllegalArgumentException on http methods

### ISSUE THREAD

**Issue by metametadata** [clj-commons/aleph#542]
```clj
(require '[aleph.http :as http])

(try
  @(http/put "https://example.com"
            {:body            123
             :read-timeout    1000
             :request-timeout 1000})

  (catch Exception e
    (prn :exception! e)))
```

## Actual

`IllegalArgumentException` is logged while returned deferred contains an error about timeout:

```
error sending body of type  java.lang.Long                                                                     aleph.http.core                      ERROR 23.06.20 21:08:45 [aleph-netty-client-event-pool-1-18] 
java.lang.IllegalArgumentException: Don't know how to convert class java.lang.Long into (stream-of io.netty.buffer.ByteBuf)
	at byte_streams$convert.invokeStatic(byte_streams.clj:196)
	at byte_streams$convert.invoke(byte_streams.clj:162)
	at aleph.netty$to_byte_buf_stream.invokeStatic(netty.clj:185)
	at aleph.netty$to_byte_buf_stream.invoke(netty.clj:184)
	at aleph.http.core$send_streaming_body.invokeStatic(core.clj:294)
	at aleph.http.core$send_streaming_body.invoke(core.clj:249)
	at aleph.http.core$eval35149$send_message__35156$fn__35157.invoke(core.clj:404)
	at aleph.http.core$eval35149$send_message__35156.invoke(core.clj:403)
	at aleph.http.client$http_connection$fn__35515$fn__35516$f__34548__auto____35517.invoke(client.clj:466)
	at aleph.http.client$http_connection$fn__35515$fn__35516$fn__35519.invoke(client.clj:465)
	at clojure.lang.AFn.run(AFn.java:22)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:465)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
:exception! #error {
 :cause "timed out after 1000 milliseconds"
 :via
 [{:type aleph.utils.RequestTimeoutException
   :message "timed out after 1000 milliseconds"
   :at [aleph.http$eval36227$request__36231$fn__36233$fn__36236$fn__36241$fn__36242 invoke "http.clj" 303]}
  {:type java.util.concurrent.TimeoutException
   :message "timed out after 1000 milliseconds"
   :at [manifold.deferred$timeout_BANG_$fn__30282 invoke "deferred.clj" 1160]}]
 :trace
 [[manifold.deferred$timeout_BANG_$fn__30282 invoke "deferred.clj" 1160]
  [manifold.time$in$f__29696 invoke "time.clj" 255]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.util.concurrent.Executors$RunnableAdapter call "Executors.java" 515]
  [java.util.concurrent.FutureTask run "FutureTask.java" 264]
  [java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask run "ScheduledThreadPoolExecutor.java" 304]
  [java.util.concurrent.ThreadPoolExecutor runWorker "ThreadPoolExecutor.java" 1128]
  [java.util.concurrent.ThreadPoolExecutor$Worker run "ThreadPoolExecutor.java" 628]
  [manifold.executor$thread_factory$reify__29358$f__29359 invoke "executor.clj" 44]
  [clojure.lang.AFn run "AFn.java" 22]
  [java.lang.Thread run "Thread.java" 834]]}
```

## Expected

`http/put` throws `IllegalArgumentException` immediately in the current thread.

## Notes

* `IllegalArgumentException` doesn't seem to be catchable even via `Thread/setDefaultUncaughtExceptionHandler` which makes it hard to detect the exception in production.
* If there are no timeouts set the returned deferred seems to wait infinitely even though there's nothing to wait for because of `IllegalArgumentException`.

**Comment by kachayev** [comment]
Interesting...

This is very specific case caused by how `send-message` detect format of the body, effectively falling back [here](https://github.com/aleph-io/aleph/blob/e8443e3d4a858ee19ad02d558c2031ff51e6f6be/src/aleph/http/core.clj#L536-L537). To solve the problem, exception from here should be propagated up to request consumption method [here](https://github.com/aleph-io/aleph/blob/e8443e3d4a858ee19ad02d558c2031ff51e6f6be/src/aleph/http/client.clj#L557-L558). Ideally, we need to make sure that anytime sending request fails
* error deferred is added into `responses` stream (otherwise the connection would just hang up until timeout)
* (probably) close the underlying connection

Let me prepare a PR to implement the logic.

**Comment by KingMob** [comment]
Fixed by #592
